tests/operators/transform/test_postgres_transform.py

```diff
@@ -0,0 +1,167 @@
+import logging
+
+import pandas as pd
+import pytest
+from airflow.models import DAG, DagRun
+from airflow.models import TaskInstance as TI
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+from airflow.utils import timezone
+from airflow.utils.session import create_session
+
+import astro.sql as aql
+from astro import dataframe as adf
+from astro.sql.table import Table, TempTable
+from tests.operators import utils as test_utils
+
+log = logging.getLogger(__name__)
+DEFAULT_DATE = timezone.datetime(2016, 1, 1)
+
+
+@pytest.fixture
+def dag():
+    return DAG(
+        "test_dag",
+        default_args={
+            "owner": "airflow",
+            "start_date": DEFAULT_DATE,
+        },
+    )
+
+
+@pytest.fixture(autouse=True)
+def cleanup():
+    yield
+    with create_session() as session:
+        session.query(DagRun).delete()
+        session.query(TI).delete()
+
+
+@adf
+def validate(df: pd.DataFrame):
+    assert len(df) == 12
+    assert df.iloc[0].to_dict()["first_name"] == "PENELOPE"
+
+
+@pytest.mark.parametrize("output_table", ["None", "partial", "full"], indirect=True)
+def test_postgres_to_dataframe_partial_output(output_table, dag):
+    @aql.transform
+    def sample_pg(input_table: Table):
+        return "SELECT * FROM {{input_table}} WHERE last_name LIKE 'G%%'"
+
+    with dag:
+        pg_output = sample_pg(
+            input_table=Table(
+                table_name="actor", conn_id="postgres_conn", database="pagila"
+            ),
+            output_table=output_table,
+        )
+        validate(df=pg_output)
+    test_utils.run_dag(dag)
+
+
+@pytest.mark.xfail
+def test_with_invalid_dag_name(sample_dag):
+    """
+    TODO: There appears to be a bug when passing an "invalid" DAG name to pandas when creating a dataframe
+    This issue can be tracked here: https://app.zenhub.com/workspaces/astro-61e7a085a496df00172965bd/issues/astro-projects/astro/212
+    :param sample_dag:
+    :return:
+    """
+    sample_dag.dag_id = "my=dag"
+
+    @aql.transform()
+    def pg_query(input_table: Table):
+        return "SELECT * FROM {{input_table}} WHERE last_name LIKE 'G%%'"
+
+    with sample_dag:
+        pg_table = pg_query(
+            input_table=Table(
+                table_name="actor", conn_id="postgres_conn", database="pagila"
+            )
+        )
+        validate(pg_table)
+    test_utils.run_dag(sample_dag)
+
+
+@pytest.fixture
+def pg_query_result(request):
```

Comment:
Why is it relevant to run all these combinations of parameters against all the databases?
Perhaps it makes sense to have some parameters that run against all databases, but not all.? Depending on the implementation, they may be redundant.